gnocchi/rest/api.py

```diff
@@ -546,19 +546,22 @@ def _lookup(self, id, *remainder):
             abort(404, six.text_type(indexer.NoSuchMetric(id)))
         return MetricController(metrics[0]), remainder
 
-    _MetricSchema = voluptuous.Schema({
-        "archive_policy_name": six.text_type,
-        "name": six.text_type,
-        voluptuous.Optional("unit"):
-            voluptuous.All(six.text_type, voluptuous.Length(max=31)),
-    })
-
     # NOTE(jd) Define this method as it was a voluptuous schema – it's just a
     # smarter version of a voluptuous schema, no?
-    @classmethod
-    def MetricSchema(cls, definition):
+    @staticmethod
+    def MetricSchema(definition):
+        creator = pecan.request.auth_helper.get_current_user(
+            pecan.request)
+
         # First basic validation
-        definition = cls._MetricSchema(definition)
+        schema = voluptuous.Schema({
+            "archive_policy_name": six.text_type,
+            "resource_id": functools.partial(ResourceID, creator=creator),
```

Comment:
this optional?

Reply:
It should be but it's not marked as such so I think it won't be. Damn, we might miss a test here.

Reply:
Ok, late reply, I forgot how this works and you confused me with "no test for /v1/metric". We have plenty of tests everywhere that create metrics in Gabbi with `POST /v1/metric` where no resource_id is set and it works. :)

Actually, all keys in voluptuous are optional except if `required=True` is passed when validating. We never do that here. The `voluptuous.Optional` flag is just used in that case, but it's not really used in any of the code currently. We usually use it for `PATCH` or `PUT` operations which are not supported on `/v1/metric`

Reply:
word. i did a bad grep.